consumer/src/main/java/com/example/consumer/consumer/KafkaConsumerService.java

```diff
@@ -0,0 +1,62 @@
+package com.example.consumer.consumer;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class KafkaConsumerService<T> {
+
+    @Autowired
+    private KafkaConsumerConfiguration configuration;
+
+    @Autowired
+    private KafkaConsumerFactory<T> kafkaConsumerFactory;
+
+    public void subscribeToTopic(
```

Comment:
Just another way how you can make things easier is to use spring-kafka features, e.g. @KafkaListener annotation, DefaultKafkaConsumerFactory for configuring consumers, etc. In that case, you can get rid of some complicated structures, while-true cycles, etc.

Reply:
updated